### ISSUE TITLE
Bug: Game crashes while building 8096x8096 and 16384x16384 sized cubemaps.

### ISSUE THREAD

**Issue by Rip-Rip-Rip** [StrataSource/Portal-2-Community-Edition#1463]
### Describe the bug

When running the "buildcubemaps" console command with 8k and 16k cubemaps the game crashes while building them.
The screenshot I have attached is the error message that occurs when trying to build the 16k cubemap. There is no error message when trying to build the 8k cubemap - the game just crashes.

![Screenshot (88)](https://github.com/StrataSource/Portal-2-Community-Edition/assets/140695311/d3ae90b0-0558-433a-9268-ff55bc4d866c)

### Issue Map

N/A (Any map with 8k or 16k cubemaps.)

### To Reproduce

1. Place a cubemap inside the map.
2. Set the "Cubemap Size" field to 14 or 15.
3. Compile and run the map.
4. Run the command "buildcubemaps".

### Operating System

Windows 11 Build 22621.1992

**Comment by JJL772** [comment]
We will probably limit cubemap size and fix this crash. Anything over 1k is insane lol

**Comment by Rip-Rip-Rip** [comment]
I agree xD, the limit I think should be 2K. 2K **could** be useful on really reflective objects like mirrors. Although 4K doesn't crash, lot's of 4k cubemaps in a level would probably inflate file sizes by a decent amount, and theres practically no visible difference between 2k and 4k cubemaps, at least at 1080p. If theres a way tho, I think there should be a way to re-enable the larger sizes and put a warning on them saying something like, this will probably crash when built, as it might just be that pc's aren't powerful enough to handle the larger sizes at the moment (I could and probably am wrong tho lol)

**Comment by JJL772** [comment]
when they're being built, cubemaps render to a RBGA32F render target. That means mip0 is:
- ~17GB for 32k
- ~4GB for 16k
- ~1GB for 8k
- ~268MB for 4k

16k and below aren't that insane actually, most GPUs have 6GB+ VRAM, modern GPUs have upwards of 20GB. That being said, 32k x 32k is the max texture size for most hardware.

**Comment by SCell555** [comment]
DX11 limit for 2D textures and cubemaps is 16k x 16k, [see docs](https://learn.microsoft.com/en-us/windows/win32/api/d3d11/ns-d3d11-d3d11_texture2d_desc). Unless you use more advanced feature, like tiled resources.
